hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/DataSourceOptions.scala

```diff
@@ -964,6 +964,11 @@ object DataSourceOptionsHelper {
 
   def translateConfigurations(optParams: Map[String, String]): Map[String, String] = {
     val translatedOpt = scala.collection.mutable.Map[String, String]() ++= optParams
+    if (!translatedOpt.contains(HoodieTableConfig.NAME.key()) &&
```

Comment:
Should this logic be incorporated into L972?

Reply:
`HoodieWriterUtils` has `sparkDatasourceConfigsToTableConfigsMap` which contains `hoodie.datasource.write.table.name` to `hoodie.table.name` mapping.  Is it not working?

Reply:
I tried adding, that, but it removes the configs with the new ones, so it was making other things fail if I used that instead

Reply:
Understood.  Should we deprioritize this for now and look for a better way?  I think it's better to have such logic in a common place instead of being scattered in a separate method, which is going to be harder to maintain.

Reply:
I don't think this makes things worse than they already are with spark configs